server/src/test/java/org/opensearch/rest/ResponseLimitSettingsTests.java

```diff
@@ -0,0 +1,221 @@
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        // Pass cluster state with 3 indices
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            clusterState.getMetadata(),
+            ResponseLimitSettings.LimitEntity.INDICES,
+            responseLimitSettings.getCatIndicesResponseLimit()
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreached_forCatShardsWithSettingDisabled_expectNotBreached() {
+        // Don't enable limit
+        final Settings settings = Settings.builder().put(CAT_SHARDS_RESPONSE_LIMIT_SETTING.getKey(), -1).build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        // Build cluster state with 3 shards
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            clusterState.getRoutingTable(),
+            ResponseLimitSettings.LimitEntity.SHARDS,
+            responseLimitSettings.getCatShardsResponseLimit()
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreached_forCatShardsWithSettingEnabled_expectBreached() {
+        // Set limit of 2 shards
+        final Settings settings = Settings.builder().put(CAT_SHARDS_RESPONSE_LIMIT_SETTING.getKey(), 2).build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        // Build cluster state with 3 shards
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            clusterState.getRoutingTable(),
+            ResponseLimitSettings.LimitEntity.SHARDS,
+            responseLimitSettings.getCatShardsResponseLimit()
+        );
+        assertTrue(breached);
+    }
+
+    public void testIsResponseLimitBreached_forCatShardsWithSettingEnabled_expectNotBreached() {
+        // Set limit of 3 shards
+        final Settings settings = Settings.builder().put(CAT_SHARDS_RESPONSE_LIMIT_SETTING.getKey(), 3).build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        // Build cluster state with 3 shards
```

Comment:
can we have one test with multiple shards per index ?

Reply:
Added multiple shards in index.